tests/simple/testdata/parse.textproto

```diff
@@ -174,3 +174,478 @@ section {
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "continue"
+    description: "Check that `continue` can be used as a struct name."
+    expr: "continue{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "else"
+    description: "Check that `else` can be used as a struct name."
+    expr: "else{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "for"
+    description: "Check that `for` can be used as a struct name."
+    expr: "for{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "function"
+    description: "Check that `function` can be used as a struct name."
+    expr: "function{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "if"
+    description: "Check that `if` can be used as a struct name."
+    expr: "if{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "import"
+    description: "Check that `import` can be used as a struct name."
+    expr: "import{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "let"
+    description: "Check that `let` can be used as a struct name."
+    expr: "let{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "loop"
+    description: "Check that `loop` can be used as a struct name."
+    expr: "loop{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "package"
+    description: "Check that `package` can be used as a struct name."
+    expr: "package{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "namespace"
+    description: "Check that `namespace` can be used as a struct name."
+    expr: "namespace{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "return"
+    description: "Check that `return` can be used as a struct name."
+    expr: "return{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "var"
+    description: "Check that `var` can be used as a struct name."
+    expr: "var{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "void"
+    description: "Check that `void` can be used as a struct name."
+    expr: "void{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "while"
+    description: "Check that `while` can be used as a struct name."
+    expr: "while{} || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+}
+section {
+  name: "struct_fields"
+  description: "Check that reserved identifiers are permitted as struct fields as long as they are not language keywords"
+  test {
+    name: "as"
+    description: "Check that `as` can be used as a struct name."
+    expr: "a{ as: 1 } || true"
```

Comment:
The recommendation would be to update the `TestAllTypes` proto with fields that use the reserved identifiers. We may need to enable backtick syntax in the CEL environment to support this case, but I believe it will work if we do.